### DISCUSSION TITLE
Do I need recent C++ and NVCC compilers to install via wheels?

### DISCUSSION THREAD

**Discussion by davidrpugh** [pyg-team/pytorch_geometric#2185] [Q&A] [answered]
Perhaps I misunderstand how installing from a wheel works but I thought that the point of installing from wheels such as those at

https://pytorch-geometric.com/whl/torch-1.7.0+cu110.html

doing so avoids the need to compile things. In particular for PyTorch Geometric users, avoids the need to compile the C++ and CUDA extensions. To install from source  GCC > 5.0 and NVCC are definitely required but I am not sure about whether these are required when installing from the provided wheels.

Do we need GCC > 5.0 and NVCC to be available in order to properly install PyTorch Geometric with GPU extensions from the provided wheels?

**Comment by rusty1s** [comment, marked as answer]
No, they are not required. However, you need a `cudatoolkit` to run the extensions with CUDA (which should already be installed via PyTorch).

**Reply by davidrpugh** [reply]
Thanks for the quick response. This is what I thought however whenever I try to install from the provided wheels the wheels fail to build because of the OS version of GCC is too old to compile the cpp extensions for PyTorch. I can fix this by adding the `cxx-compiler` meta-package from `conda-forge` but I didn't think anything needed to be compiled when using wheels.

I am most concerned about confirming that the CUDA extensions have been compiled properly. Is there any import statement that will fail in the event that the CUDA extensions have not been compiled?

**Reply by rusty1s** [reply]
Can you try to run `pip` with the `--no-index` flag?
```
pip install --no-index torch-scatter -f ...
```
This will ensure that installing is done via the provided wheels.

**Reply by davidrpugh** [reply]
This seems to work. Is there any quick way to confirm that they CUDA extensions have installed properly? Maybe an import statement or something that will fail if CUDA extensions have not been compiled?

**Reply by rusty1s** [reply]
Importing the package without any error indicates that CUDA extensions have been loaded properly. Alternatively, you can also run our test suites to confirm that.

**Reply by davidrpugh** [reply]
Great! Thanks for your help.
